clib/src/keys.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright © 2021 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use librad::{
+    keys::{IntoSecretKeyError, PublicKey, SecretKey},
+    profile::Profile,
+    signer::BoxedSigner,
+};
+use radicle_keystore::{
+    crypto::{Crypto, KdfParams, Pwhash, SecretBoxError},
+    file,
+    pinentry::Prompt,
+    FileStorage,
+    Keystore as _,
+};
+
+/// The filename for storing the secret key.
+pub const LIBRAD_KEY_FILE: &str = "librad.key";
+
+pub type Error = file::Error<SecretBoxError<std::io::Error>, IntoSecretKeyError>;
+
+/// Create a [`Prompt`] for unlocking the key storage.
+pub fn prompt() -> Pwhash<Prompt<'static>> {
+    let prompt = Prompt::new("please enter your passphrase: ");
+    Pwhash::new(prompt, KdfParams::recommended())
```

Comment:
In upstream there's a feature flag to switch this to test KDF params. Do we want to have a similar thing here for when you're testing CLI applications? Especially when testing CLI applications against test Upstream sandboxes.